model/textparse/protobufparse.go

```diff
@@ -480,3 +508,18 @@ func isNativeHistogram(h *dto.Histogram) bool {
 		h.GetZeroCount() > 0 ||
 		h.GetZeroThreshold() > 0
 }
+
+// isFloatHistogram checks that processed histogram is a float histogram if it
+// some of the histogram defined with float values
+func isFloatHistogram(h *dto.Histogram) bool {
+	floatBuckets := false
+	for _, b := range h.Bucket {
+		if b.CumulativeCountFloat > 0 {
+			floatBuckets = true
+			break
+		}
+	}
```

Comment:
The buckets are likely in `NegativeCount` and `PositiveCount`. The comments say `Bucket` is for conventional histograms and not the new native histograms. In which case I think it should be as simple as checking of length of NegativeCount or PositiveCount is >0.

Reply:
Yeah, the (new) `cumulative_count_float` is just there for consistency so that we can also have a conventional float histogram (which is already working with the old Prometheus text format, but (strictly speaking) invalid in OpenMetrics and impossible in the old Prometheus protobuf format).

It would be nice to also add the code to parse conventional float histograms, but that doesn't have to happen in this PR.

For this PR, we are mainly interested in the new native AKA sparse histograms. So you have to ignore the `Bucket` proto message and just look if `NegativeCount` or `PositiveCount` have any elements, as @codesome said.